### ISSUE TITLE
[MusicBERT]: How to fill masked tokens in an input sequence after training?

### ISSUE THREAD

**Issue by aspil** [microsoft/muzic#57]
Hello again,

I have fine-tuned MusicBERT on masked language modeling using a custom dataset. I have loaded the fine-tuned checkpoint using:
```
roberta = RobertaModel.from_pretrained( # MusicBERTModel.from_pretrained also works
    '.',
    checkpoint_file=sys.argv[1],
    data_name_or_path=sys.argv[2],
    user_dir='musicbert'
)
```
What I want to do is to give it an input sequence, mask one or more tokens before passing the input to the model and somehow predict them. Something like masked language modeling, but with control over which tokens I want to mask and predict.

What I cannot understand is what format the input sequence should be in order to be passed to the model, and how to make the model predict the masked tokens in the input.
I have tried to replicate it by looking at the fairseq's training code since I want to do something similar, but it's too complicated.

Thanks in advance.

**Comment by tripathiarpan20** [comment]
Hi!

Logically it should be the same as [fairseq RoBERTa mask filling example](https://github.com/facebookresearch/fairseq/blob/main/examples/roberta/README.md#filling-masks), but after reading #59 (I just posted it at the time of commenting this), you may see why that might not be possible yet. 

FYI: The mask token according to `roberta.task.label_dictionary` is 1236: 

![image](https://user-images.githubusercontent.com/42506819/177048690-b71073c4-2ee7-4019-8082-516e7a1a9a32.png)

**Comment by mlzeng** [comment]
Hi @aspil 

The mask filling example seems won't work for MusicBERT tokens. You can replace some tokens in the input tensor with the <mask> token (the value is 1236), get the output tensor from the LM head for the prediction, and use argmax to get the index of the predicted tokens.

**Comment by tripathiarpan20** [comment]
Hi @mlzeng 

I have implemented customized masking functions and easy interface for users to perform remixing of MIDIs using MusicBERT base with random masking of chosen instruments and many other parameters in this [notebook](https://colab.research.google.com/drive/1C7jS-s1BCWLXiCQQyvIl6xmCMrqgc9fg?usp=sharing), described further in README of my [repo](https://github.com/tripathiarpan20/midiformers).

It would be helpful for interested users in the future to have a link to this notebook (or) README of the repo in `musicbert/README.md` to interact with the model with their favorite MIDIs. 

Let me know if this is possible to do.'
Thanks.

**Comment by mlzeng** [comment]
Hi @tripathiarpan20 

The MIDI remixing tool works well. It's a brilliant project! I have added the links in `README.md`.

**Comment by aspil** [comment]
Hello @mlzeng and @tripathiarpan20 
I'll be closing this issue as it is more than resolved. The MIDIformers project is indeed brilliant!
